packages/expo-dev-menu/app/components/DevIndicator.tsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import { View, ViewProps } from 'react-native';
+
+type Props = ViewProps & {
+  isActive: boolean;
+  isNetworkAvailable: boolean;
+};
+
+export default class DevIndicator extends React.PureComponent<Props> {
+  render() {
+    let backgroundColor;
+    if (this.props.isActive && this.props.isNetworkAvailable) {
+      backgroundColor = '#00c100';
+    } else if (!this.props.isNetworkAvailable) {
+      backgroundColor = '#e0e057';
+    } else {
+      backgroundColor = '#ccc';
+    }
+
+    return (
+      <View
+        style={[
+          {
+            width: 7,
+            height: 7,
+            backgroundColor,
+            borderRadius: 3.5,
+          },
+          this.props.style,
+        ]}
+      />
+    );
+  }
+}
```

Comment:
https://reactjs.org/docs/react-api.html#reactmemo 😉 
Less lines, I'm happier 😂 
```suggestion
const DevIndicator: React.FC<Props> = React.memo(({ isActive, isNetworkAvailable, style }) => {
  const backgroundColor = isActive && isNetworkAvailable
    ? '#00c100'
    : !isNetworkAvailable
    ? '#e0e057'
    : '#ccc';
 
  return (
    <View style={[styles.indicator, { backgroundColor }, style]} />
  );
});

const styles = StyleSheet.create({
  indicator: {
    width: 7,
    height: 7,
    borderRadius: 3.5,
  },
});
```